### DISCUSSION TITLE
Memory requirement for runs

### DISCUSSION THREAD

**Discussion by pat-s** [renovatebot/renovate#24123] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.68.0

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I wanted to ask what the minimum memory requirement is for renovate runs? I've configured `renovoate` to run on a self-hosted CI in k8s and had to give it 2GB of memory to not get killed during a run (previous step was 1.5GB which got killed).

The run has the following stats

```
       "stats": {
         "managers": {"terraform": {"fileCount": 67, "depCount": 127}},
         "total": {"fileCount": 67, "depCount": 127}
       }
```

- Is this memory footprint expected?
- Does it grow even more if more files are scanned?

### What you tried so far.

-

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
Speaking for the hosted Renovate app, we operated all servers with 3.75GB of RAM without noticing any problems. So I'd regard 2GB to be well within expectations.

Considering how cheap server RAM is and the value which Renovate gives, I'm not concerned with proactively improving memory utilization at this point, although PRs are welcome if users find ways to overcome inefficiencies. We don't currently have any issues open of people reporting problematic memory leaks either. 

BTW Renovate executes many types of package manager (npm, Gradle, pip, etc) and has no control/estimate over how much memory they take up.

**Reply by pat-s** [reply]
Thanks for the reply! My comment was less aiming for "it should take less" but rather "what to expect" given that I am defining resource definitions in the k8s environment for its runs.

I am aware that it invokes other package managers and hence it's footprint might vary.
But knowing to be within the ballpark of 2-4GB for a run already helps a lot. Maybe this piece of information can somehow find its way into the docs :)
